### PR TITLE
Automate Hook Smoke Test

### PR DESCRIPTION
**Pull request by Klovely**
#### Describe Bug or Feature
Automate Hook smoke Test : 
1. Test create and print a hook
2. Test import and export a hook
3. Test enable and disable a hook
4.  Test modify a hook
5. Test delete a hook

#### Describe Your Change
Added script to cover above tests

#### Attach Test and Valgrind Logs/Output
* [Test_execution_logs2.txt](https://github.com/PBSPro/pbspro/files/3456273/Test_execution_logs2.txt)
* [Test_execution_logs_sles15.txt](https://github.com/PBSPro/pbspro/files/3456793/Test_execution_logs_sles15.txt)

### CODE REVIEW

**Comment by Klovely** [comment]
Latest execution logs :- 
[Test_execution_logs_latest.txt](https://github.com/PBSPro/pbspro/files/3470709/Test_execution_logs_latest.txt)

**Comment by Klovely** [comment]
Addressed Dinesh's offline comment as well .

**Comment by Klovely** [comment]
Thanks @prakashcv13 and @dinesh1306 for comments.
Incorporated your comments in latest commits, please have a look.

File: test/tests/functional/pbs_hooksmoketest.py [hunk 1]
```diff
@@ -84,31 +80,30 @@ def clean_hooks(self):
...
-            self.server.manager(MGR_CMD_DELETE, HOOK, id="my_hook_queuejob")
-        if 'my_hook_queuejob2' in self.server.hooks:
-            self.server.manager(MGR_CMD_DELETE, HOOK, id="my_hook_queuejob2")
-        if 'my_hook' in self.server.hooks:
-            self.server.manager(MGR_CMD_DELETE, HOOK, id="my_hook")
+        del_hook = ['test_hook_queuejob',
+                    'test_hook_queuejob2', 'test_hook2', 'test_hook']
+        for i in del_hook:
+            if i in self.server.hooks:
+                self.server.manager(MGR_CMD_DELETE, HOOK, id=i)
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: this is unnecessary. we are looping through the list just to find the one hook that was created in a test.
We can use only one name, and delete that hook in tearDown()

File: test/tests/functional/pbs_hooksmoketest.py [hunk 2]
```diff
@@ -19,25 +19,21 @@
...
         TestFunctional.setUp(self)
-        self.server.manager(MGR_CMD_SET, SERVER, {'log_events': 2047})
-        self.server.manager(MGR_CMD_SET, SERVER,
-                            {'scheduling': 'False'})
+        a = {'log_events': 2047, 'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
         self.clean_hooks()
 
-    def get_file_path(self, hook_name):
+    def get_file_paths(self, hook_name, existence=False):
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: now, the function does the job of checking the .hk file, so we should name it accordingly.

File: test/tests/functional/pbs_hooksmoketest.py [hunk 3]
```diff
@@ -19,25 +19,21 @@
...
     """
-
     def setUp(self):
         TestFunctional.setUp(self)
-        self.server.manager(MGR_CMD_SET, SERVER, {'log_events': 2047})
-        self.server.manager(MGR_CMD_SET, SERVER,
-                            {'scheduling': 'False'})
+        a = {'log_events': 2047, 'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
         self.clean_hooks()
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: why do we need to delete the hooks in setUp()?

File: test/tests/functional/pbs_hooksmoketest.py [hunk 4]
```diff
@@ -49,32 +45,32 @@ def get_file_path(self, hook_name):
...
-                                path=self.hk_file_location,
+                                path=hk_file_location,
                                 sudo=True)
             count = count - 1
             if rc or count == 0:
                 break
             time.sleep(1)
         self.assertEqual(rc, existence, _msg)
         self.logger.info(msg)
+        return server_hooks_dir, hk_file, hk_file_location
```
Comment (prakashcv13) [inline test/tests/functional/pbs_hooksmoketest.py]: the function does not need to return the paths anymore, the caller is not using them.

**Comment by Klovely** [comment]
Latest execution logs:- 
[Test_execution_logs_latest_after_change.txt]
(https://github.com/PBSPro/pbspro/files/3471989/Test_execution_logs_latest_after_change.txt)

**Comment by Klovely** [comment]
Latest execution logs:-
===============
[Test_execution_logs_latest_4.txt](https://github.com/PBSPro/pbspro/files/3475626/Test_execution_logs_latest_4.txt)

**Comment by Klovely** [comment]
Thanks @prakashcv13  , as per our offline discussion.
Incorporated your comments in latest commits, please have a look.

**Comment by Klovely** [comment]
Thanks @prakashcv13  and @dinesh1306 , as per our offline discussion.
Incorporated your comments in latest commits, please have a look.

**Comment by Klovely** [comment]
Latest execution logs :- 
[Test_execution_logs_latest_5.txt](https://github.com/PBSPro/pbspro/files/3476073/Test_execution_logs_latest_5.txt)

**Comment by prakashcv13** [comment]
I am alright with the changes. Will merge once the checks finish.

**Comment by Klovely** [comment]
All check passed .
Please take this PR forward.

**Pull request merged**